src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -184,19 +184,116 @@ public void DrawGeometry(IBrush brush, IPen pen, IGeometryImpl geometry)
             }
         }
 
+        struct BoxShadowFilter : IDisposable
+        {
+            public SKPaint Paint;
+            private SKImageFilter _filter;
+            public SKClipOperation ClipOperation;
+
+            public static BoxShadowFilter Create(BoxShadow shadow, double opacity, bool skipDilate)
```

Comment:
BoxShadow should except an existing SKPaint so this can reuse an existing one like CreatePaint